src/pkg/provisioner/provisioner.go

```diff
@@ -29,12 +29,24 @@ import (
 	"golang.org/x/sys/unix"
 )
 
-func setup(dockerCredentialGCR string, systemd *systemdClient) error {
+// I typically do not like this style of mocking, but I think it's the best
+// option in this case. These functions cannot execute at all in a normal test
+// environment because they require root privileges. Even if the address to
+// mount is owned by the caller, these functions will fail. To take them out of
+// the test codepath, we can mock them here.
+//
+// I have considered an alternative involving writing a unixPkg interface and
+// passing it through the Deps struct. But it doesn't give us much for its
+// additional verbosity.
```

Comment:
makes sense! I agree injecting those through the Deps structure does not hold any purpose of its own in the non-testing code.